kolibri/plugins/user_profile/tasks.py

```diff
@@ -0,0 +1,50 @@
+from django.core.management import call_command
+from rest_framework import serializers
+
+from kolibri.core.auth.models import FacilityUser
+from kolibri.core.auth.tasks import PeerImportSingleSyncJobValidator
+from kolibri.core.auth.utils.migrate import merge_users
+from kolibri.core.serializers import HexOnlyUUIDField
+from kolibri.core.tasks.decorators import register_task
+from kolibri.core.tasks.job import Priority
+from kolibri.core.tasks.permissions import IsAdmin
+from kolibri.core.tasks.permissions import NotProvisioned
+
+
+class MergeUserValidator(PeerImportSingleSyncJobValidator):
+    local_user_id = serializers.PrimaryKeyRelatedField(
+        queryset=FacilityUser.objects.all()
+    )
+    local_user_id = HexOnlyUUIDField()
+
+    def validate(self, data):
+        job_data = super(MergeUserValidator, self).validate(data)
+        local_user_id = data["local_user_id"]
+        if not FacilityUser.objects.filter(id=local_user_id).exists():
+            raise serializers.ValidationError(
+                "An user with the given id does not exist in this facility"
+            )
+        job_data["kwargs"]["local_user_id"] = local_user_id
+        return job_data
+
+
+@register_task(
+    queue="soud",
+    validator=MergeUserValidator,
+    priority=Priority.HIGH,
+    cancellable=False,
+    track_progress=True,
+    permission_classes=[IsAdmin() | NotProvisioned()],
```

Comment:
ah, yes, my mistake. I left Not_Provisioned there because I was testing the setup wizard too. Of course, it does not make sense. For the IsSelf, I agree, that's a good idea. I've implemented it